src/datalad_installer.py

```diff
@@ -2084,7 +2084,7 @@ def install_package(
 
     def assert_supported_system(self, **kwargs: Any) -> None:
         if not (ON_LINUX or ON_MACOS or ON_WINDOWS):
-            raise MethodNotSupportedError(f"{SYSTEM} OS not supported")
+            raise MethodNotSupportedError(f"{SYSTEM} OS not supported by {self}")
```

Comment:
I think these should be `{self.NAME}` instead, as that shows the name used on the command line.

Reply:
ok, let's do that

```suggestion
            raise MethodNotSupportedError(f"{SYSTEM} OS not supported by {self.NAME}")
```
?

while at it -- is there a way to include NAME of the component component (git-annex, etc) in addition?

Reply:
You would have to pass `component` to `assert_supported_system()` on line 1468 and update all the `assert_supported_system()` definitions to include the string in the error message.

Reply:
ok, let's then skip that for now  -- too much for `sed` ;)